migration/src/main/java/org/corfudb/migration/EndpointMigration.java

```diff
@@ -0,0 +1,79 @@
+package org.corfudb.migration;
+
+import static com.google.common.collect.Sets.newHashSet;
+import static org.corfudb.migration.LogFormat1to2.getChecksum;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.Set;
+
+public class EndpointMigration {
+
+    /**
+     * Data-store files responsible for layout recovery.
+     */
+    private static final Set<String> layoutStateRecoveryFiles =
+            newHashSet("LAYOUT_CURRENT", "MANAGEMENT_LAYOUT");
+
+    /**
+     * Modifies the endpoint in the layout datastore files.
+     *
+     * @param args Accepts the arguments in the following order,
+     *             dir:         CorfuDB data directory,
+     *             oldEndpoint: Old address the server was binding to,
+     *             newEndpoint: New address to which the server will bind to.
+     * @throws Exception if migration fails.
+     */
+    public static void main(String[] args) throws Exception {
+        if (args.length < 3) {
+            throw new IllegalArgumentException("Expected parameters: CorfuDB data directory,"
+                    + "old endpoint and new endpoint");
+        }
+
+        String corfuDir = args[0];
+        String oldEndpoint = args[1];
+        String newEndpoint = args[2];
+
+        // Transform the layout state local datastore files.
+        modifyBindingIpAddress(corfuDir, oldEndpoint, newEndpoint);
+    }
+
+    /**
+     * Modifies the ip address in the layout files.
+     *
+     * @param dirStr      Directory path
+     * @param oldEndpoint Old endpoint to be replaced.
+     * @param newEndpoint New endpoint.
+     */
+    static void modifyBindingIpAddress(String dirStr,
+                                       String oldEndpoint,
+                                       String newEndpoint) throws IOException {
+        File dir = new File(dirStr);
+        File[] layoutStateFiles = dir.listFiles(
+                (dir1, name) -> layoutStateRecoveryFiles.stream().anyMatch(name::startsWith));
+
+        for (File file : layoutStateFiles != null ? layoutStateFiles : new File[0]) {
```

Comment:
I think this can be simplified with an if statement instead of a 0-length array ?

Reply:
Replaced with if statement.